src/metrics.ts

```diff
@@ -182,33 +182,53 @@ export const MetricsProcessor = (
     };
   };
 
-  const _send = (): void => {
+  const _send = async () => {
     if (closed) {
       return;
     }
 
     const metrics: Metrics = _summarize();
     if (metrics) {
       log.debug('Start sending metrics data');
-      api
-        .postMetrics(environment, cluster, metrics)
-        .then((response) => {
-          log.debug('Metrics server returns: ', response.status);
-          infoMetricsSuccess(log);
-          if (response.status >= 400) {
-            log.error(
-              'Error while sending metrics data with status code: ',
-              response.status,
-            );
-          }
-        })
-        .catch((error: Error) => {
-          warnPostMetricsFailed(`${error}`, log);
-          log.debug('Metrics server returns error: ', error);
-        });
+      try {
+        const response = await postMetricsWithRetry(
```

Comment:
Just out of curiosity, why change from using the Promise methods to async/await?

Reply:
I think it was just an after effect of hacking to get it working correctly, and I inadvertantly changed it to async/await. No other reason. But it loos like the only use of a promise method in this file - so at least it didn't break the pattern.